lib/Runtime/Library/JavascriptLibrary.cpp

```diff
@@ -6929,6 +6933,30 @@ namespace Js
         return function;
     }
 
+    JavascriptPromiseThenFinallyFunction* JavascriptLibrary::CreatePromiseThenFinallyFunction(JavascriptMethod entryPoint, RecyclableObject* OnFinally, RecyclableObject* Constructor, bool shouldThrow)
+    {
+        Assert(scriptContext->GetConfig()->IsES6PromiseEnabled());
+        
+        FunctionInfo* functionInfo = RecyclerNew(this->GetRecycler(), FunctionInfo, entryPoint);
+        DynamicType* type = CreateDeferredPrototypeFunctionType(entryPoint);
```

Comment:
Can you use `GetDeferredAnonymousFunctionTypeHandler()` to avoid having to set the `name` property here?

Reply:
How did I not notice that function... I spent 10-15 minutes searching JavascriptLibrary.cpp for a way to do that but didn't find it. Changed.

Reply:
Note, this comment has moved down to the other CreateDeferredPrototypeFunctionType call not where you originally placed it.